packages/node/example/multiSession/src/serverSideApp.ts

```diff
@@ -0,0 +1,164 @@
+
+import express from "express";
+const app = express();
+const PORT = 3001;
+
+const DEFAULT_OIDC_ISSUER = "https://broker.pod.inrupt.com/";
+
+app.use(
+  cookieSession({
+    name: "session",
+    // These keys are required by cookie-session to sign the cookies.
+    keys: ["key1", "key2"],
+    maxAge: 24 * 60 * 60 * 1000, // 24 hours
+  })
+);
+
+app.get("/", async (req, res, next) => {
+  const sessionIds = await getSessionIdFromStorageAll();
+  const sessions = await Promise.all(
+    sessionIds.map(async (sessionId) => {
+      return await getSessionFromStorage(sessionId);
+    })
+  );
+  const htmlSessions =
+    sessions.reduce((sessionList, session) => {
+      if (session?.info.isLoggedIn) {
+        return sessionList + `<li>${session?.info.webId}</li>`;
+      }
+      return "Anonymous";
+    }, "<ul>") + "</ul>";
+  res
+    .writeHead(200, { "Content-Type": "text/html" })
+    .write(
+      `<p>There are currently ${sessionIds.length} visitors: ${htmlSessions}</p>`
+    );
+  res.end();
+});
+
+app.get("/login", async (req, res, next) => {
+  const session = new Session();
+  req.session!.sessionId = session.info.sessionId;
+  await session.login({
+    redirectUrl: "http://localhost:3001/redirect",
+    oidcIssuer: DEFAULT_OIDC_ISSUER,
+    clientName: clientApplicationName,
+    handleRedirect: (data) => {
+      res.writeHead(302, {
+        location: data,
+      });
+      res.end();
+    },
+  });
+  if (session.info.isLoggedIn) {
+    res
+      .writeHead(200, { "Content-Type": "text/html" })
+      .write(`<p>Already logged in with WebID: [${session.info.webId}]</p>`);
+    res.end();
+  }
+});
+
+app.get("/redirect", async (req, res) => {
+  const session = await getSessionFromStorage(req.session!.sessionId);
+  if (session === undefined) {
+    res
+      .writeHead(400, { "Content-Type": "text/html" })
+      .write(`<p>No session stored for ID ${req.session!.sessionId}</p>`);
+  } else {
+    await session.handleIncomingRedirect(getRequestFullUrl(req));
+    if (session.info.isLoggedIn) {
+      res
+        .writeHead(200, { "Content-Type": "text/html" })
+        .write(`<p>Logged in as [${session.info.webId}] after redirect</p>`);
+    } else {
+      res
+        .writeHead(400, { "Content-Type": "text/html" })
+        .write(`<p>Not logged in after redirect</p>`);
+    }
+  }
+  res.end();
+});
+
+app.get("/fetch", async (req, res, next) => {
+  const session = await getSessionFromStorage(req.session!.sessionId);
+  if (!req.query["resource"]) {
+    res
+      .writeHead(400, { "Content-Type": "text/html" })
+      .write("<p>Expected a 'resource' query param</p>");
+  } else if (session) {
+    res
+      .writeHead(200, { "Content-Type": "text/html" })
+      .write(
+        "<pre>" +
+          (
+            await (await session.fetch(req.query["resource"] as string)).text()
+          ).replace(/</g, "&lt;") +
+          "</pre>"
```

Comment:
Maybe it'd be better to just pipe the return value of `fetch` directly to the user, rather than injecting it into an HTML page?

Reply:
Because of the `<url>` that are interpreted as HTML, Turtle looks quite bad if it isn't processed a bit.

Reply:
But why have it interpreted as HTML? Can't we just pass on the Content-Type as well?

Reply:
Oh yeah I see what you mean, I missed your point initially. Unfortunately, for `Content-Type: text/turtle`, my browser doesn't display it, it opens a popup to open it with an application. i guess it could be configured otherwise, but for simplicity think wrapping the content in `<pre>` tags is still a good option.

Reply:
Yeah, certainly for demo purposes, wrapping in HTML makes the most sense. But I think maybe Vinnie is referring to this script being **_actually useful_** - i.e. where you want the script to fetch a private resource for you, and return that resource as pure RDF (or binary if it's a Non-RDFSource). That's a really good point (if that's what Vinnie meant, otherwise it might just be a silly point that I'm making :) !) - so I'd support that ability with a separate `/fetchRaw` endpoint that does just stream back whatever the server replies with. (But not for now - in a subsequent PR)